File: csharp/sum-of-multiples/SumOfMultiples.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max) => 
        Enumerable.Range(1, max - 1)
            .Where(x => multiples.Any(n => x % n == 0))
            .Sum();
}

File: fsharp/sum-of-multiples/SumOfMultiples.fs
module SumOfMultiples

open System

let multiples (max: int) (n: int) : int list = 
    [ for i in n .. n .. max -> i ]

let sum (numbers: int list) (upperBound: int): int = 
    numbers
    |> List.collect (multiples (upperBound - 1))
    |> Seq.distinct
    |> Seq.sum 